client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -131,9 +131,17 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
             request.withHeader(header.name(), header.value());
         }
 
+        String mimeType = request.headers().value("Content-Type");
```

Comment:
I think we should talk about this line. This says to me that we're allowing our service/swagger interfaces (I think I'm just going to call them service interfaces from now on. That's what we call them in our code) to set Content-Type as a header. The problem I have with this is that Content-Type is tied very closely to the body, and it's kind of a big deal. If the Content-Type is set incorrectly, then the target server might not function correctly (the same goes for the "Accept" header, but that's a different discussion). Since we're the client libraries, I'm leaning towards us always setting the Content-Type header based on the type of the body argument passed to the function.
So, in short, I agree with the changes you've made on lines 136 through 144, but I'm not sure how I feel about this line. Let's leave it here for now, though.

Reply:
Yeah, I think we should drill down to the root of how/why `@Headers(“Content-Type: ...”)` is being generated by AutoRest and work back to something reasonable.